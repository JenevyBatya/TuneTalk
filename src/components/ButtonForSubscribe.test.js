import { render, screen, fireEvent } from "@testing-library/react";
import SubscribeButton from "./ButtonForSubscribe";
import "@testing-library/jest-dom";

describe("Компонент SubscribeButton", () => {
    test("отображает кнопку с начальным текстом 'Подписаться'", () => {
        render(<SubscribeButton />);
        const button = screen.getByRole("button", { name: /подписаться/i });
        expect(button).toBeInTheDocument();
        expect(button).toHaveTextContent("Подписаться");
    });

    test("изменяет текст на 'Подписаны' при клике", () => {
        render(<SubscribeButton />);
        const button = screen.getByRole("button", { name: /подписаться/i });
        fireEvent.click(button);
        expect(button).toHaveTextContent("Подписаны");
    });

    test("переключает текст обратно на 'Подписаться' при повторном клике", () => {
        render(<SubscribeButton />);
        const button = screen.getByRole("button", { name: /подписаться/i });
        fireEvent.click(button);
        fireEvent.click(button);
        expect(button).toHaveTextContent("Подписаться");
    });

});
