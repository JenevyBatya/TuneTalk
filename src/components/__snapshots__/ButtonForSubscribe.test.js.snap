// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeButton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-3fw551"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1ygm10f-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Подписаться
    </button>
  </div>
</DocumentFragment>
`;

exports[`Компонент SubscribeButton соответствует сохранённому снимку 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-3fw551"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1ygm10f-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Подписаться
    </button>
  </div>
</DocumentFragment>
`;
